### DISCUSSION TITLE
Question: does mapping function in solidity automatically create an array?

### DISCUSSION THREAD

**Discussion by vicshi06** [smartcontractkit/full-blockchain-solidity-course-py#797] [Q&A] [answered]
Does mapping function in solidity automatically create an array? And that's how we are able to retrieve data from a mapping function?

**Comment by cromewar** [comment, marked as answer]
Hello @vicshi06 mappings are it's own type and they are kind of arrays but not fully, I suggest you to check [this](https://medium.com/upstate-interactive/mappings-in-solidity-explained-in-under-two-minutes-ecba88aff96e) to understand deeply how mappings works.
